safedelete/queryset.py

```diff
@@ -90,10 +91,12 @@ def _filter_visibility(self):
         This is because QuerySet._fetch_all cannot work with a clone.
         """
         force_visibility = getattr(self, '_safedelete_force_visibility', None)
+        filter_applied = getattr(self, '_safedelete_filter_applied', False)
         visibility = force_visibility \
             if force_visibility is not None \
             else self._safedelete_visibility
-        if visibility in (DELETED_INVISIBLE, DELETED_VISIBLE_BY_FIELD, DELETED_ONLY_VISIBLE):
+        if visibility in (DELETED_INVISIBLE, DELETED_VISIBLE_BY_FIELD, DELETED_ONLY_VISIBLE) \
+           and not filter_applied:
```

Comment:
Can we check for `not filter_applied` first to avoid a checking a tuple? Now that I think of it as well, can we make that tuple a set to slightly improve performance?
```python
{DELETED_INVISIBLE, DELETED_VISIBLE_BY_FIELD, DELETED_ONLY_VISIBLE}
```